_pytest/python.py

```diff
@@ -335,6 +337,9 @@ class PyCollector(PyobjMixin, pytest.Collector):
     def funcnamefilter(self, name):
         return self._matches_prefix_or_glob_option('python_functions', name)
 
+    def isnosetestfunction(self, obj):
+        return getattr(obj, '__test__', False)
```

Comment:
`test_parsefactories_evil_objects_issue214` is failing, which shows that in order to be safe during collection (#214), this would be safer:

```python
def isnosetestfunction(self, obj):
    try:
        return obj.__test__
    except:
        return False
```

(I don't like using bare `except:` clauses, but I think this is the only option here).